invenio_app_ils/circulation/api.py

```diff
@@ -5,117 +5,146 @@
 
-"""Invenio App ILS Circulation apis."""
+"""Invenio App ILS Circulation APIs."""
 
 import uuid
+from copy import deepcopy
 
-from dateutil import parser
+from elasticsearch import VERSION as ES_VERSION
 from flask import current_app
-from invenio_circulation.api import Loan, patron_has_active_loan_on_item
+from invenio_circulation.api import Loan
 from invenio_circulation.pidstore.minters import loan_pid_minter
 from invenio_circulation.proxies import current_circulation
+from invenio_circulation.search.api import search_by_patron_item_or_document
 from invenio_db import db
 
-from invenio_app_ils.errors import InvalidParameterError, \
-    MissingRequiredParameterError, PatronHasLoanOnDocumentError, \
-    PatronHasLoanOnItemError
+from invenio_app_ils.errors import MissingRequiredParameterError, \
+    PatronHasLoanOnItemError, PatronHasRequestOnDocumentError
 from invenio_app_ils.proxies import current_app_ils_extension
 from invenio_app_ils.records.api import Item
 
 
-def request_loan(params):
-    """Create a loan and trigger the first transition to create a request."""
-    if "patron_pid" not in params:
-        raise MissingRequiredParameterError(
-            description="'patron_pid' is required on loan request")
-    if "document_pid" not in params:
-        raise MissingRequiredParameterError(
-            description="'document_pid' is required on loan request")
+def _validate_delivery(delivery):
+    """Validate `delivery` param."""
+    methods = list(
+        current_app.config.get("CIRCULATION_DELIVERY_METHODS", {}).keys()
+    )
+    if methods:
+        if not delivery or delivery["method"] not in methods:
+            raise MissingRequiredParameterError(
+                description="A valid 'delivery' is required on loan request"
+            )
 
-    if patron_has_active_loan_on_item(patron_pid=params["patron_pid"],
-                                      document_pid=params["document_pid"]):
-        raise PatronHasLoanOnDocumentError(params["patron_pid"],
-                                           params["document_pid"])
 
-    _ensure_delivery_method_provided(params)
+def _set_item_to_can_circulate(item_pid):
+    """Change the item status to CAN_CIRCULATE."""
+    item = Item.get_record_by_pid(item_pid)
+    if item["status"] != "CAN_CIRCULATE":
+        item = item.patch(
+            [{"op": "replace", "path": "/status", "value": "CAN_CIRCULATE"}]
+        )
+        item.commit()
+        db.session.commit()
+        current_app_ils_extension.item_indexer.index(item)
+
+
+def patron_has_request_on_document(patron_pid, document_pid):
+    """Return True if patron has a request for the given document."""
+    search = search_by_patron_item_or_document(
+        patron_pid=patron_pid,
+        document_pid=document_pid,
+        filter_states=current_app.config["CIRCULATION_STATES_LOAN_REQUEST"],
+    )
+    search_result = search.execute()
+    if ES_VERSION[0] >= 7:
```

Comment:
Could you reuse this convention [here](https://github.com/inveniosoftware/invenio-records-rest/blob/2e87d7bf8466e99487f6d2a0ddb6e91e17033bcc/invenio_records_rest/serializers/json.py#L19) just to have something consistent with the changes we did along invenio?